src/include/sched_cmds.h

```diff
@@ -43,53 +43,42 @@
 extern "C" {
 #endif
 
+#include "pbs_ifl.h"
 typedef struct sched_cmd sched_cmd;
-typedef struct sched_svrconn sched_svrconn;
 
 struct sched_cmd {
 	/* sched command */
 	int cmd;
 
 	/* jobid assisiated with cmd if any else NULL */
 	char *jid;
-
-	/*
-	 * connection to server from which we got this sched command.
-	 * This will be one of the server's secondary connection.
-	 *
-	 * This will be used in while finding server conn structure before calling
-	 * schedule() which needs primary connection to the server
-	 *
-	 */
-	int from_sock;
-};
-struct sched_svrconn {
-	char *svrhost;	    /* address of host in host[:port] format */
-	int primary_sock;   /* primary connection to server - used for IFL calls */
-	int secondary_sock; /* secondary connection to server - used to receive sched cmds from server */
 };
 
 /* server to scheduler commands: */
-#define SCH_ERROR         -1
-#define SCH_SCHEDULE_NULL  0
-#define SCH_SCHEDULE_NEW   1		/* New job queued or eligible	*/
-#define SCH_SCHEDULE_TERM  2		/* Running job terminated	*/
-#define SCH_SCHEDULE_TIME  3		/* Scheduler interval reached	*/
-#define SCH_SCHEDULE_RECYC 4		/* Not currently used		*/
-#define SCH_SCHEDULE_CMD   5		/* Schedule on command 		*/
-#define SCH_CONFIGURE      7
-#define SCH_QUIT           8
-#define SCH_RULESET        9
-#define SCH_SCHEDULE_FIRST 10		/* First schedule after server starts */
-#define SCH_SCHEDULE_JOBRESV 11		/* Arrival of an existing reservation time */
-#define SCH_SCHEDULE_AJOB  12		/* run one, named job */
-#define SCH_SCHEDULE_STARTQ 13		/* Stopped queue started */
-#define SCH_SCHEDULE_MVLOCAL 14		/* Job moved to local queue */
-#define SCH_SCHEDULE_ETE_ON 15		/* eligible_time_enable is turned ON */
-#define SCH_SCHEDULE_RESV_RECONFIRM 16		/* Reconfirm a reservation */
-#define SCH_SCHEDULE_RESTART_CYCLE 17		/* Restart a scheduling cycle */
+enum svr_sched_cmd
+{
+	SCH_ERROR = -1,
```

Comment:
As this command is not being used scheduler does not receive it. So we don't store it in sched_cmds table so there is no problem. Since there is no usage of this command I have now removed it in server/sched code.

Reply:
Sure, remove it.  We could probably remove a few more if we wanted.  Originally (like >20 yrs ago), the idea was that scheduling was site dependent, so each site would write their own scheduler.  We provided pbs_sched.c which would link into the sites scheduler.  They would provide 'schedinit()' and schedule()' which we would call.  The sched commands which were defined were a guess at what would be useful.  When I wrote the scheduler, I chose the commands I thought were useful, and turned the rest  into a noop.